pandas/core/generic.py

```diff
@@ -3789,6 +3790,18 @@ def _take(
 
         See the docstring of `take` for full explanation of the parameters.
         """
+        if not isinstance(indices, slice):
+            indices = np.asarray(indices, dtype=np.intp)
+            if (
+                axis == 0
+                and indices.ndim == 1
+                and using_copy_on_write()
+                and array_equal_fast(
+                    indices,
+                    np.arange(0, len(self), dtype=np.intp),
```

Comment:
Further idea to potentially speed this up: right now we only use `array_equal_fast` to check an array against a standard 0...n indexer, I think? 
For that specific case, we don't actually need to create this second array, but could just use the iteration variable inside `array_equal_fast`. 

(now, I don't know if we would want to start using `array_equal_fast` for other cases)

Reply:
(that also avoids creating this additional array even for the fast cases where the length doesn't even match)

Reply:
I’ll take a look at this in a follow up if ok to check how big the performance improvement would be

Reply:
Sure